torch/jit/_recursive.py

```diff
@@ -67,12 +69,31 @@ def copy_to_script_module(original, stubs):
             continue
         item = getattr(original, name)
         if name in class_annotations:
+            if is_optional_module_annotation(class_annotations[name]):
+                # See [optional module]
+                continue
             the_type = torch.jit.annotations.ann_to_type(class_annotations[name])
         else:
             the_type = torch._C._jit_try_infer_type(item)
         if the_type is not None:
             script_module._c._register_attribute(name, the_type, item)
 
+    for name in class_annotations:
+        annotation = class_annotations[name]
```

Comment:
Why can't the body of this traversal be inlined under the if in the earlier loop?